binstall download url for riscv64gc target does not resolve correctly

I'm trying out the public [RISC-V GitHub runners hosted by RISE](https://riseproject.dev/2026/03/24/announcing-the-rise-risc-v-runners-free-native-risc-v-ci-on-github/) for building the [Mullvad VPN app](https://github.com/MarkusPettersson98/mullvadvpn-app-riscv/) natively for RISC-V in CI, and as part of that I want to install `cargo-deb` and `cargo-generate-rpm`, but when trying to install `cargo-binstall`  the wrong download URL is resolved. `riscv64` is used as the architecture instead of `riscv64gc` : https://github.com/MarkusPettersson98/mullvadvpn-app-riscv/actions/runs/23742454447/job/69162621433

```bash
Run cargo-bins/cargo-binstall@main
Run set -eu
+ set -o pipefail
+ set -o pipefail
+ case "${BINSTALL_VERSION:-}" in
++ mktemp -d
+ cd /tmp/tmp.32e346K5YE
+ '[' -z '' ']'
+ base_url=https://github.com/cargo-bins/cargo-binstall/releases/latest/download/cargo-binstall-
++ uname -s
+ os=Linux
+ '[' Linux = Darwin ']'
+ '[' Linux = Linux ']'
++ uname -m
+ machine=riscv64
+ '[' riscv64 = armv7l ']'
+ target=riscv64-unknown-linux-musl
+ '[' riscv64 = armv7 ']'
+ url=https://github.com/cargo-bins/cargo-binstall/releases/latest/download/cargo-binstall-riscv64-unknown-linux-musl.tgz
+ do_curl https://github.com/cargo-bins/cargo-binstall/releases/latest/download/cargo-binstall-riscv64-unknown-linux-musl.tgz
+ curl --retry 10 -A 'Mozilla/5.0 (X11; Linux x86_64; rv:60.0) Gecko/20100101 Firefox/81.0' -L --proto =https --tlsv1.2 -sSf https://github.com/cargo-bins/cargo-binstall/releases/latest/download/cargo-binstall-riscv64-unknown-linux-musl.tgz
+ tar -xvzf -
curl: (22) The requested URL returned error: 404

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
Error: Process completed with exit code 2.
```

What's resolved: https://github.com/cargo-bins/cargo-binstall/releases/latest/download/cargo-binstall-riscv64-unknown-linux-musl.tgz
Should resolve to: https://github.com/cargo-bins/cargo-binstall/releases/latest/download/cargo-binstall-riscv64gc-unknown-linux-musl.tgz

--- a/install-from-binstall-release.sh
+++ b/install-from-binstall-release.sh
@@ -37,6 +37,9 @@ elif [ "$os" = "Linux" ]; then
     if [ "$machine" = "armv7l" ]; then
         machine="armv7"
     fi
+    if [ "$machine" = "riscv64" ]; then
+        machine="riscv64gc"
+    fi
     target="${machine}-unknown-linux-musl"
     if [ "$machine" = "armv7" ]; then
         target="${target}eabihf"

